fvm/transactionStorageLimiter_test.go

```diff
@@ -126,25 +127,32 @@ func accountExists(owner string) OwnerKeyValue {
 	}
 }
 
-func newMockState(updatedKeys []string, ownerKeyStorageValue []OwnerKeyValue) *state.State {
+func newMockStateHolder(updatedKeys []string, ownerKeyStorageValue []OwnerKeyValue) *state.StateHolder {
 
-	ledger := state.NewMapLedger()
-	s := state.NewState(ledger)
+	view := utils.NewSimpleView()
+	s := state.NewState(view)
+	sm := state.NewStateHolder(s)
 
 	for _, okv := range ownerKeyStorageValue {
-		_ = s.Set(okv.Owner, "", okv.Key, utils.Uint64ToBinary(okv.Value))
+		_ = s.Set(okv.Owner, "", okv.Key, uint64ToBinary(okv.Value))
 	}
-	_ = s.Commit()
 
 	for _, key := range updatedKeys {
 		_ = s.Set(key, "", "", []byte("1"))
 	}
 
-	return s
+	return sm
 }
 
-func mockGetStorageCapacityFuncFactory(_ *fvm.VirtualMachine, _ fvm.Context, _ *fvm.TransactionProcedure, _ *state.State, _ *fvm.Programs) (func(address common.Address) (value uint64, err error), error) {
+func mockGetStorageCapacityFuncFactory(_ *fvm.VirtualMachine, _ fvm.Context, _ *fvm.TransactionProcedure, _ *state.StateHolder, _ *fvm.Programs) (func(address common.Address) (value uint64, err error), error) {
 	return func(address common.Address) (value uint64, err error) {
 		return 100, nil
 	}, nil
 }
+
+// uint64ToBinary converst a uint64 to a byte slice (big endian)
+func uint64ToBinary(integer uint64) []byte {
```

Comment:
we have a duplicate implementation of this same function. Maybe it is better to encapsulate it exportable at as common tool.

Reply:
Yeah I intentionally did this to remove the dependency to ledger package, I'll move them into utils in the next PR